packages/@aws-cdk/cdk/lib/token.ts

```diff
@@ -35,6 +35,19 @@ export class Token {
     return unresolved(obj);
   }
 
+  /**
+   * Retrieves the `IResolvable` for of an encoded token.
+   *
+   * @param encodedToken an encoded Token.
+   *
+   * @returns the resolvable object for the given encoded token, or `undefined` if `encodedToken` is not an encoded
+   *          token (`Token.isUnresolved(encodedToken)` is `false`), or the token is a `string` composed of the
```

Comment:
If token is a string that is NOT concatenated.

I don't like this function being here. You should be using the `reverseString` function instead, and deal with the case that the string could be concatenated. This API being here just opens it up to being misused (because it's more "convenient" than the other one) and creating subtle bugs.